### DISCUSSION TITLE
searchable

### DISCUSSION THREAD

**Discussion by ribbbo** [filamentphp/filament#12335] [Help] [answered]
### Package

Table builder

### Package Version

v2

### How can we help you?

Searching the encrypted value in TextColumn
TextColumn::make('full_name')
    ->searchable(query: function (Builder $query, string $search): Builder {
        return $query
            ->where('first_name', 'like', "%{$search}%")
            ->where('last_name', 'like', "%{$search}%");
    })

**Comment by leandrocfe** [comment]
Is it a question?

**Comment by micraux** [comment]
Do you want to have a research based on first name _**or**_ last name?

If so, you second `->where()` should be `->orWhere()`:

```php
TextColumn::make('full_name')
    ->searchable(query: function (Builder $query, string $search): Builder {
        return $query
            ->where('first_name', 'like', "%{$search}%")
            ->orWhere('last_name', 'like', "%{$search}%");
})
```

**Reply by micraux** [reply]
@ribbbo, you closed this thread as resolved.

Did you solve it by yourself or did you successfully use the solution I gave?

If it is the second one, I would really appreciate a `Mark as answer` ;-)

**Comment by ribbbo** [comment, marked as answer]
I solved it my self. Sorry for the confusion.
My problem is searching the field that is encrypted stored in database
the solution I use is like this
TextColumn::make('full_name')
    ->searchable(query: function (Builder $query, string $search): Builder {
    $datas = Model::get()->toArray();
    $ids = [];
    foreach ($datas as $data){
      $strpos = strpos($data['full_name'], $eas
      $ids[] = $data['id'];
    }
        return $query->where('id', $ids);
})
